integration_tests/auth/test_auth.py

```diff
@@ -1,658 +1,583 @@
+        u = _get_user(role_or_user)
+        with _EXPECTATION_401:
+            _patch_viewer(None, u.password, new_username="new_username")
+
+    @pytest.mark.parametrize("role_or_user", [_MEMBER, _ADMIN, _CZAR])
+    def test_cannot_change_password_without_current_password(
+        self,
+        role_or_user: _RoleOrUser,
+        _get_user: _GetUser,
+        _passwords: Iterator[_Password],
+    ) -> None:
+        u = _get_user(role_or_user)
+        doer = u.log_in()
+        new_password = next(_passwords)
+        with pytest.raises(Exception):
+            _patch_viewer(doer, None, new_password=new_password)
+
+    @pytest.mark.parametrize("role_or_user", [_MEMBER, _ADMIN, _CZAR])
+    def test_cannot_change_password_with_wrong_current_password(
+        self,
+        role_or_user: _RoleOrUser,
+        _get_user: _GetUser,
+        _passwords: Iterator[_Password],
+    ) -> None:
+        u = _get_user(role_or_user)
+        doer = u.log_in()
+        assert (wrong_password := next(_passwords)) != doer.password
+        new_password = next(_passwords)
+        with pytest.raises(Exception):
+            _patch_viewer(doer, wrong_password, new_password=new_password)
 
     @pytest.mark.parametrize("role", list(UserRoleInput))
-    def test_user_can_change_password_for_self(
+    def test_change_password(
         self,
         role: UserRoleInput,
-        patch_viewer: _PatchViewer,
-        log_in: _LogIn,
-        get_new_user: _GetNewUser,
-        passwords: Iterator[_Password],
+        _get_user: _GetUser,
+        _passwords: Iterator[_Password],
     ) -> None:
-        user = get_new_user(role)
-        email = user.profile.email
-        password = user.profile.password
-        token = user.token
-        new_password = f"new_password_{next(passwords)}"
-        assert new_password != password
-        wrong_password = next(passwords)
-        assert wrong_password != password
-        for _token, _password in product((None, token), (None, wrong_password, password)):
-            if _token == token and _password == password:
-                continue
-            with pytest.raises(BaseException):
-                patch_viewer(_token, _password, new_password=new_password)
-            log_in(password, email=email).__enter__()
-        patch_viewer((old_token := token), (old_password := password), new_password=new_password)
-        another_password = f"another_password_{next(passwords)}"
-        with pytest.raises(HTTPStatusError, match="401 Unauthorized"):
-            patch_viewer(old_token, new_password, new_password=another_password)
-        with pytest.raises(HTTPStatusError, match="401 Unauthorized"):
-            log_in(old_password, email=email).__enter__()
-        new_token, _ = log_in(new_password, email=email).__enter__()
-        with pytest.raises(BaseException):
-            patch_viewer(new_token, old_password, new_password=another_password)
+        u = _get_user(role)
+        doer = u.log_in()
```

Comment:
```suggestion
        user = _get_user(role)
        user = user.log_in()
```

Terminology of `doer` is not clear.

Reply:
`u` is chosen because it's a homophone of `you`. It's also acting like `I` as being the protagonist of the test. it's distinguished from doer, because it's not logged in, so multiple doers are possible for the same `u`

Reply:
> u is chosen because it's a homophone of you

FWIW I don't think this is apparent from reading the tests. I though it was just a shortened version of `user`.

Reply:
> it was just a shortened version of user

that's also true, haha

Reply:
changed `doer` to `logged_in_user`